site_scons/site_tools/roc/pkgconfig.py

```diff
@@ -0,0 +1,70 @@
+import os
+
+def FindPkgConfig(env, toolchain):
+    if env.HasArg('PKG_CONFIG'):
+        return True
+
+    if toolchain:
+        if env.Which(toolchain + '-pkg-config'):
+            env['PKG_CONFIG'] = toolchain + '-pkg-config'
+            return True
+        if env.Which('PKG_CONFIG_PATH') \
+            or env.Which('PKG_CONFIG_LIBDIR') \
+            or env.Which('PKG_CONFIG_SYSROOT_DIR'):
```

Comment:
`env.Which(foo)` -> `foo in os.environ`

(earlier I suggested HasArg, but that's incorrect; they should be in environment to be available for pkg-config and can't be scons arguments)

Reply:
ok! I removed PKG_CONFIG_PATH in the override arg then. 
does `foo in os.environ` mean `foo in env.Dictionary()`?

Reply:
I guess
```python
def FindPkgConfigPath(context):
    env = context.env

    context.Message("Searching PKG_CONFIG_PATH...")

    if env.HasArg('PKG_CONFIG_PATH'):
        context.Result(env['PKG_CONFIG_PATH'])
        return True
```
also needs to change to `'PKG_CONFIG_PATH' in os.environ` or `'PKG_CONFIG_PATH' in env.Dictionary()`?

Reply:
I think no. HasArg() checks that the var is either in os.environ or passed to scons as an argument, like:

```
scons ... VAR=value
```

Both ways are ok for us, and both ways we'll get the value of PKG_CONFIG_PATH and use it for installation rule. So here we should use HasArg.

However, in this check:

```
        if 'PKG_CONFIG_PATH' in os.environ \
            or 'PKG_CONFIG_LIBDIR' in os.environ \
            or 'PKG_CONFIG_SYSROOT_DIR' in os.environ:
            if env.Which('pkg-config'):
```

we want to ensure that PKG_CONFIG_PATH (or other vars) is passed to `pkg-config` tool. Environment variables are passed, but scons arguments aren't. So here we should check only for environment variables. If the user declared them, likely pkg-config will behave correctly with the cross-compilation sysroot and likely we can use it without asking the user to specify it explicitly.

Note that this is not a completely strict check, just a reasonable default. I expect that many cross-compilation environments have those vars exported, and when they are, our build script will just work out of the box. However, if our detection will go awry, the user can always specify PKG_CONFIG_PATH and PKG_CONFIG manually.

Reply:
TLDR: the current checks look good :)

Reply:
But since we use HasArg for PKG_CONFIG_PATH, we should return PKG_CONFIG_PATH to this loop:

```
for var in ['CXX', 'CC', 'AR', 'RANLIB', 'RAGEL', 'GENGETOPT', 'PKG_CONFIG', 'CONFIG_GUESS']:
    env.OverrideFromArg(var)
```

IIRC, we need OverrideFromArg if we want to allow passing a var as an argument. OverrideFromArg checks whether a var was specified as an argument (`scons ... VAR=...`), and if it was, it stores it into `env`.

Reply:
The rest is good!

Reply:
thank you! I added `PKG_CONFIG_PATH` back to the override arg list.